src/core/unit.c

```diff
@@ -207,94 +206,129 @@ static void unit_init(Unit *u) {
                 UNIT_VTABLE(u)->init(u);
 }
 
+static int unit_add_another_name(Unit *u, char *name) {
+        _cleanup_set_free_ Set *set = NULL;
+        int r;
+
+        /* Make sure that u->names is allocated. If name is non-null, add it to the set. */
+
+        if (!u->names) {
+                set = set_new(&string_hash_ops);
+                if (!set)
+                        return -ENOMEM;
+
+                /* We convert from state where u->id is the only name to a state
+                 * where u->names is allocated and u->id points into it. */
+
+                r = set_put(set, u->id);
+                if (r < 0)
+                        return r;
+                assert(r > 0);
+        }
+
+        if (name) {
+                set_put(set ?: u->names, name);
+                if (r < 0)
+                        return r;
+                assert(r > 0);
+        }
+
+        if (!u->names)
+                u->names = TAKE_PTR(set);
+        return 0;
+}
+
 int unit_add_name(Unit *u, const char *text) {
-        _cleanup_free_ char *s = NULL, *i = NULL;
+        _cleanup_free_ char *name = NULL, *inst = NULL;
```

Comment:
why not spell "inst" out? I first though this was about "Installing" rather than "instance". no need to save characters here